src/Strategy/Node/Npm/PackageLock.hs

```diff
@@ -54,8 +54,25 @@ import Graphing (Graphing)
 import Path (Abs, File, Path)
 import Strategy.Node.PackageJson (Development, FlatDeps (devDeps, directDeps), NodePackage (pkgName), Production)
 
-newtype NpmPackageJson = NpmPackageJson
-  {packageDependencies :: Map Text NpmDep}
+data NpmPackagesPkg = NpmPackagesPkg
+  { pkgPeerDeps :: Map Text Text
+  , pkgResolved :: Maybe Text
+  }
+  deriving (Eq, Ord, Show)
+
+instance FromJSON NpmPackagesPkg where
+  parseJSON = withObject "NpmPackagesPkg" $ \obj ->
+    -- Note that the object with a key of "" represents the current project we're
+    -- analyzing. It has a different set of keys/values.
+    NpmPackagesPkg
+      <$> obj .:? "peerDependencies" .!= Map.empty
+      <*> obj .:? "resolved"
+
+data NpmPackageJson = NpmPackageJson
+  { packageDependencies :: Map Text NpmDep
+  , -- |  Data from the "packages" field of package-json.lock
```

Comment:
I don't think this comment adds anything.  The `FromJSON` instance tells us where this comes from, and this whole module is focused solely on `package-lock.json`.

Reply:
I've renamed some things, please let me know if you think the names make sense. I've tried to mirror in the hierarcy of the file in the names.